lib/coverband/integrations/middleware.rb

```diff
@@ -9,7 +9,11 @@ def initialize(app)
     def call(env)
       @app.call(env)
     ensure
-      Coverband::Collectors::Coverage.instance.report_coverage
+      if Coverband.configuration.background_reporting_enabled
+        Coverband::Background.start
+      else
+        Coverband::Collectors::Coverage.instance.report_coverage
```

Comment:
Do we even want to keep this option around?

Reply:
yeah see other comments, I'd rather not get rid of the middleware flow until we ensure on all setups the background thread works well. Feels like it would be good to have both as an option until we get a bit more real-world testing and feedback on the new background flow.

Reply:
Agreed.